geoopt/utils.py

```diff
@@ -36,3 +36,13 @@ def make_tuple(obj):
         return (obj,)
     else:
         return obj
+
+
+def idx2sign(idx, dim, neg=True):
+    if neg:
+        if idx < 0:
+            return idx
+        else:
+            return (idx + 1) % -(dim + 1)
```

Comment:
Some tldr and less nesting levels

Reply:
If you have positive indexing of dims, this will make the proper negative index that is used in broadcasting. I will delete that